src/Agent/NewRelic/Agent/Core/DistributedTracing/TracingState.cs

```diff
@@ -254,5 +248,43 @@ public static ITracingState AcceptDistributedTraceHeaders<T>(T carrier, Func<T,
 
             return tracingState;
         }
+
+        /// <summary>
+        /// Use remote parent sampled behavior configuration in conjunction with the traceparent sampled flag to determine
+        /// if the transaction should be sampled.
+        /// </summary>
+        private void ApplyRemoteParentSampledBehavior(
+            RemoteParentSampledBehavior remoteParentSampledBehavior,
+            RemoteParentSampledBehavior remoteParentNotSampledBehavior)
+        {
+            // don't do anything if the traceparent is not present or if behavior is configured to default
+            if (!_traceContext.TraceparentPresent ||
+                (remoteParentNotSampledBehavior == RemoteParentSampledBehavior.Default &&
+                 remoteParentSampledBehavior == RemoteParentSampledBehavior.Default))
+            {
+                return;
+            }
+
+            var sampledBehavior = _traceContext.Traceparent.Sampled
+                ? remoteParentSampledBehavior
+                : remoteParentNotSampledBehavior;
+
+            switch (sampledBehavior)
+            {
+                case RemoteParentSampledBehavior.AlwaysOn:
+                    _priority = 2.0f; // per the spec, set priority high so that this sample is always kept
+                    _sampled = true;
+                    break;
+                case RemoteParentSampledBehavior.AlwaysOff:
+                    _priority = 0.0f; // set lowest possible priority
+                    _sampled = false;
+                    break;
+                // don't need a case for Default, as it is handled above
+                default:
+                    throw new ArgumentException($"Invalid {(_traceContext.Traceparent.Sampled ? "remoteParentSampledBehavior" : "remoteParentNotSampledBehavior")} value: {sampledBehavior}.");
+            }
+
+            Log.Finest($"ApplyRemoteParentSampledBehavior:  _traceContext.Traceparent.Sampled={_traceContext.Traceparent.Sampled}, {(_sampled.Value ? "remoteParent" : "remoteParentNot")}SampledBehavior: {sampledBehavior} ==> Sampled: {Sampled}, Priority:{Priority}");
```

Comment:
The string formatting for this finest level log message can be deferred if the code is switched to the `Log.Finest("sampled={sampledValue}", sampledValue)` style of log message strings.

Reply:
I always forget that. We probably should add a ticket to the backlog to scrub "improper" log formatting - it's used in many, many places.